asset-transfer-basic/chaincode-go/go.mod

```diff
@@ -2,8 +2,6 @@ module github.com/hyperledger/fabric-samples/asset-transfer-basic/chaincode-go
 
 go 1.21
 
-toolchain go1.22.4
```

Comment:
I think I caused this when I bulk updated dependencies to the v2 chaincode APIs. By default the `go mod` command sets the toolchain to the specific Go version is used to run the command, which might impose more restrictive version constraints than the `go` directive.

In [other repositories](https://github.com/hyperledger/fabric-gateway/commit/798c6ef971820ea461c2e8226dcbe55613c12cae) we have been taking the approach of avoiding a `toolchain` directive but specifying a full (3-digit) version for the `go` directive. For current Go versions, the `go` version is used as the `toolchain` if no toolchain is specified, and toolchain must be a full (3-digit) Go version. That might be the right thing to do here too.